src/app/components/LocationMap/MapwithMarker.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useRef, useEffect, useState } from 'react';
+import mapboxgl from 'mapbox-gl';
+import type { Map } from 'mapbox-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+import styled from 'styled-components';
+import MapboxGeocoder from '@mapbox/mapbox-gl-geocoder';
+import '@mapbox/mapbox-gl-geocoder/dist/mapbox-gl-geocoder.css';
+
+type MapWithMarkerProps = {
+  displayArea: string;
+};
+
+export default function MapWithMarker({
+  displayArea,
+}: MapWithMarkerProps): JSX.Element {
+  if (typeof import.meta.env.VITE_MAPBOX_ACCESS_KEY === 'string') {
+    mapboxgl.accessToken = import.meta.env.VITE_MAPBOX_ACCESS_KEY;
+  } else {
+    throw new Error('no KEY provided');
+  }
+
+  const mapContainer = useRef<HTMLDivElement | null>(null);
+  const map = useRef<null | Map>(null);
+  const [longitude] = useState<number>(10.0125);
+  const [latitude] = useState<number>(53.5469);
+  const [zoom] = useState(9);
+  const marker = new mapboxgl.Marker();
+
+  const geocoder = new MapboxGeocoder({
+    accessToken: mapboxgl.accessToken,
+    marker: false,
+  });
+
+  geocoder.on('result', (result) => {
+    marker.remove;
+    const markerPoint = result.result.center;
+    map.current ? marker.setLngLat(markerPoint).addTo(map.current) : null;
+    map.current?.flyTo({ center: markerPoint });
+  });
```

Comment:
You create an event handler on each render. 
Instead, I would create a single `geocoder` on mount (`useEffect`).